integration/cli_test.go

```diff
@@ -999,8 +1000,14 @@ func TestNodeCommand(t *testing.T) {
 	assert.Equal(t, "node-4", listAll[3].GetName())
 	assert.Equal(t, "node-5", listAll[4].GetName())
 
-	assert.Equal(t, "TestOS", listAll[0].GetHostInfo().AsMap()["OS"])
-	assert.Equal(t, "DebugTestNode", listAll[0].GetHostInfo().AsMap()["Hostname"])
+	assert.NotNil(t, listAll[0].TailCfg)
+
+	var hi *tailcfg.Hostinfo
+	err = json.Unmarshal([]byte(listAll[0].TailCfg.HostInfo), &hi)
```

Comment:
By this point we should make sure its already unmarshalled and part of the JSON that is presented to the user, not an escaped string.

Reply:
> By this point we should make sure its already unmarshalled and part of the JSON that is presented to the user, not an escaped string.

The user will be provided with tailcfg.HostInfo serialized into a string that will be passed to gRPC, resulting in JSON inside JSON. And I understand that this is exactly what you assumed, and if so, then I don't quite understand what your phrase means. The user himself will have to parse the JSON from the string.
> its already unmarshalled and part of the JSON that is presented to the user